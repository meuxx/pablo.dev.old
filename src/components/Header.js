import React from 'react'
import Img from 'gatsby-image'
import { StaticQuery, graphql } from 'gatsby'

import Footer from './Footer'

export default () => {
  const avatarQuery = graphql`
    query {
      avatar: file(relativePath: { eq: "avatar.jpg" }) {
        childImageSharp {
          # Specify the image processing specifications right in the query.
          # Makes it trivial to update as your page's design changes.
          fluid(maxWidth: 256) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `

  return (
    <header id="header">
      <div className="inner">
        <StaticQuery
          query={avatarQuery}
          render={data => (
            <Img
              fluid={data.avatar.childImageSharp.fluid}
              className="image avatar"
              title="Avatar"
            />
          )}
        />
        <h1>
          <strong>Pablo Maceda</strong>
          <br /> <a href="mailto:contact@pablo.dev">contact@pablo.dev</a>
        </h1>
        <br />
        <h2>
          Back-end developer at{' '}
          <a
            href="https://www.kooomo.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            kooomo.com
          </a>
          <br /> London, UK
        </h2>
      </div>
      <Footer />
    </header>
  )
}
