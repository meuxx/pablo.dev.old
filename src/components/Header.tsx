import React, { useMemo } from 'react'

import Avatar from './Avatar'
import Footer from './Footer'
import { Sidebar, SidebarMain, SidebarTitle, SidebarSubtitle, Link } from '../styles/styles'
import useFathomGoal from '../hooks/useFathomGoal'
import data from '../data.json'

const getShortUrl = (url: string): string => {
  try {
    const { hostname } = new URL(url)
    const domain = hostname.replace(/^(www\.)?/, '')
    return `${domain.charAt(0).toUpperCase()}${domain.slice(1)}`
  } catch (e) {
    return url
  }
}

const Header: React.FC = () => {
  const handleMailGoal = useFathomGoal('NMFGPZ35')
  const handleOrganizationGoal = useFathomGoal('HBXPURPT')

  const shortUrl = useMemo(() => getShortUrl(data.author.organizationUrl), [])

  return (
    <Sidebar>
      <SidebarMain>
        <Avatar />

        <SidebarTitle>
          <strong>{data.site.author}</strong>
          <br />{' '}
          <Link
            href={`mailto:${data.author.email}`}
            rel="noopener noreferrer"
            target="_blank"
            aria-label="Contact email"
            title="Contact email"
            onClick={handleMailGoal}
          >
            {data.author.email}
          </Link>
        </SidebarTitle>

        <br />

        <SidebarSubtitle>
          {data.author.jobTitle} at{' '}
          <Link
            href={data.author.organizationUrl}
            rel="noopener noreferrer"
            target="_blank"
            title={data.author.organizationName}
            onClick={handleOrganizationGoal}
          >
            {shortUrl}
          </Link>
          <br />
          {`${data.author.city}, ${data.author.country}`}
        </SidebarSubtitle>
      </SidebarMain>

      <Footer />
    </Sidebar>
  )
}

export default Header
