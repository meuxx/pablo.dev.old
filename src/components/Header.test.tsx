import React from 'react'
import { render } from '@testing-library/react'
import Header from './Header'

jest.mock('./Avatar', () => () => <div />)
jest.mock('./Footer', () => () => <div />)

describe('Header', () => {
  it('renders without crashing', () => {
    const { container } = render(<Header />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
