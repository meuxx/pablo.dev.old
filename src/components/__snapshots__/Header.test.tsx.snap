// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
.c4 {
  -webkit-transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  border-bottom: dotted 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c4:hover {
  border-bottom-color: transparent;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #033E6B;
  background-attachment: scroll,fixed;
  background-image: url(test-file-stub);
  background-repeat: repeat;
  color: rgba(255,255,255,0.5);
  height: 100%;
  left: 0;
  padding: 4em 2em;
  position: fixed;
  text-align: right;
  top: 0;
  width: 24%;
}

.c0 > * {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0 0 1em 0;
}

.c3 {
  font-weight: 700;
  color: rgba(255,255,255,0.5);
  font-size: 1.35em;
  line-height: 1.75em;
  margin: 0;
}

.c3 strong {
  font-weight: 700;
  color: #ffffff;
}

.c5 .c2  @media screen and (max-width:980px) {
  font-size: 1.35em;
}

@media screen and (max-width:1280px) {
  .c0 {
    padding: 6em 3em 3em 3em;
    width: 30%;
  }
}

@media screen and (max-width:980px) {
  .c0 {
    background-attachment: scroll;
    background-position: top left,center center;
    background-size: auto,cover;
    left: auto;
    padding: 6em 4em;
    position: relative;
    text-align: center;
    top: auto;
    width: 100%;
    min-height: 100vh;
    display: block;
  }
}

@media screen and (max-width:736px) {
  .c0 {
    padding: 2.25em 1.5em;
  }
}

@media screen and (max-width:480px) {
  .c0 {
    padding: 4.5em 1.5em;
  }
}

@media screen and (max-width:1280px) {
  .c3 {
    font-size: 1.25em;
  }
}

@media screen and (max-width:980px) {
  .c3 {
    font-size: 1.75em;
  }
}

@media screen and (max-width:736px) {
  .c3 {
    font-size: 1.35em;
  }
}

@media screen and (max-width:736px) {
  .c5 {
    font-size: 1.35em;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div />
      <h1
        class="c2 c3"
      >
        <strong>
          Pablo Maceda
        </strong>
        <br />
         
        <a
          aria-label="Contact email"
          class="c4"
          href="mailto:contact@pablo.dev"
          rel="noopener noreferrer"
          target="_blank"
          title="Contact email"
        >
          contact@pablo.dev
        </a>
      </h1>
      <br />
      <h2
        class="c5"
      >
        Software Engineer
         at
         
        <a
          class="c4"
          href="https://www.sainsburys.co.uk/"
          rel="noopener noreferrer"
          target="_blank"
          title="Sainsbury's"
        >
          Sainsburys.co.uk
        </a>
        <br />
        London, UK
      </h2>
    </div>
    <div />
  </header>
</div>
`;
